search_util.go

```diff
@@ -0,0 +1,592 @@
+//  Copyright (c) 2023 Couchbase, Inc.
```

Comment:
Why this unrelated change of moving all this content from search.go into search_util.go. Undo this.

Reply:
We can look into better arranging this content later if needed.